app/views/Settings/LocationTrackingStatus.js

```diff
@@ -0,0 +1,184 @@
+import styled, { css } from '@emotion/native';
+import React, { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { Linking } from 'react-native';
+import DeviceSettings from 'react-native-device-settings';
+import { openSettings } from 'react-native-permissions';
+
+import { Button, Divider, Switch, Typography } from '../../components';
+import Colors from '../../constants/colors';
+import { config } from '../../COVIDSafePathsConfig';
+import { useLocTrackingStatus } from '../../services/hooks/useLocTrackingStatus';
+import { Reason } from '../../services/LocationService';
+import { isPlatformiOS } from '../../Util';
+
+export const LocationTrackingStatus = () => {
+  const { i18n } = useTranslation();
+  const [locTrackingStatus, setLocTrackingStatus] = useLocTrackingStatus();
+  console.log(locTrackingStatus);
+
+  const { reason, canTrack } = locTrackingStatus;
+
+  const isGPS = config.tracingStrategy === 'gps';
+
+  const hasLocTrackingPermissions =
```

Comment:
Can this just be the inverse of the isMissingPermissions?

Reply:
So the way that it works now is that since `setUserOptIn()` is async, it is initially `undefined`.

So we essentially have three return cases:

1. User has permissions
2. User does not have permissions
3. Waiting on the permission status to resolve

In cases 1 & 2, we return a component. In case 3, we don't want to return anything or else we end up with a flicker between the button and the switch components if the state is different once resolved.

Feels hacky/unclear though so open to feedback on how to make it clear.

Reply:
I think this is a good solution 

```js
const getGpsStatusElement = () => {
    if (reason === undefined) {
      return;
    }

    if (hasPermission) {
      return (
        <GpsStatusSwitch canTrack={canTrack} setUserOptIn={setUserOptIn} />
      );
    } else {
      return <EnableGpsButton reason={reason} />;
    }
  };
```